keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -140,6 +141,23 @@ data class ClusterSpec(
   val defaults: ServerGroupSpec
     @JsonUnwrapped get() = _defaults
 
+  @JsonIgnore
+  override val artifactType: ArtifactType? = ArtifactType.deb
+
+  // Provides a hint as to cluster -> artifact linkage even _before_ resolution has had a chance to run.
+  override val artifactName: String?
+    @JsonIgnore get() = when (imageProvider) {
+      is ArtifactImageProvider -> imageProvider.deliveryArtifact.name
+      else -> null
+    }
+
+  // Provides a hint as to cluster -> artifact linkage even _before_ resolution has had a chance to run.
+  override val artifactReference: String?
+    @JsonIgnore get() = when (imageProvider) {
+      is ReferenceArtifactImageProvider -> imageProvider.reference
+      else -> null
+    }
```

Comment:
Artifact name and reference are now dynamically determined from the image provider defined in the spec.